chapters/packages.tex

```diff
@@ -220,9 +220,12 @@ \subsection{Directory Hierarchy Mapping}\label{mapping-a-package-class-hierarchy
 A directory shall not contain both the sub-directory \filename{A} and the file \filename{A.mo}.
 \end{example}
 
-In order to preserve the order of sub-entities it is advisable to create a file \filename{package.order} where each line contains the name of one class or constant (using its Modelica \lstinline!IDENT! form).
-If a \filename{package.order} is present when reading a structured entity the classes and constants are added in this order; if the contents does not exactly match the classes and constants in the package, the resulting order is tool specific and a warning may be given.
+In order to preserve the order of sub-entities it is advisable to create a file \filename{package.order} where each line contains the name of one class, constant (using its Modelica \lstinline!IDENT! form), or \lstinline!extends!-clause (in the form \lstinline!extends A.B! -- one space, no semicolon).
+If a \filename{package.order} is present when reading a structured entity the classes, constants, and optionally \lstinline!extends!-clauses are added in this order; if the contents does not exactly match the classes and constants in the package, the resulting order is tool specific and a warning may be given.
+The relative order of classes, constants, and optionally \lstinline!extends!-clauses that are stored in \filename{package.mo} must be consistent with the relative order of the corresponding entries in \filename{package.order} (that is, the \filename{package.order} only adds information regarding the ordering of classes not stored in \filename{package.mo}).
 Classes and constants that are stored in \filename{package.mo} are also present in \filename{package.order} but their relative order should be identical to the one in \filename{package.mo} (this ensures that the relative order between classes and constants stored in different ways is preserved).
+An \lstinline!extends!-clause entry shall be omitted from \filename{package.order} if it would appear before any class not stored in \filename{package.mo}.
```

Comment:
This rule is so complicated that I'm afraid few humans will ever learn it, and by not even allowing the obvious natural thing of adding every `extends`-clause, there is no migration path to a simple logic that even a human could remember.  Shouldn't we strive for something simpler that also humans can learn as the main rule, and use deprecation to still permit the complicated rule?

Reply:
My order of preference is thus:
1. shall or should, I can live with either.
2. reject proposal
3. "may as deprecated behavior"
4. may

Reply:
Off-line poll.

Reply:
> The downside by deprecating the current behavior is that we force all libraries to be upgraded, and that will be too much pain for very little benefit, as packages regularly have extends-clauses, but rarely want them later. I would say that not changing package.order is preferable to that.

The point of deprecation is that it does _not_ force all libraries to be upgraded anytime soon.  We could have it deprecated for ten years, effectively just recommending to do it the other way for the foreseeable future.  Then, when we have given libraries ample time to adapt, we can remove the deprecated behavior completely.